waspc/packages/wasp-config/__tests__/mapUserSpecToAppSpecDecls.unit.test.ts

```diff
@@ -0,0 +1,1103 @@
+    const auth = Fixtures.getAuth('full')
+    testMapAuth(auth, {
+      overrideRoutes: [Fixtures.getRoute('email-verification').name],
+      shouldError: true,
+    })
+  })
+
+  function testMapAuth(
+    auth: UserApi.AuthConfig,
+    options:
+      | {
+          overrideEntities?: string[]
+          overrideRoutes?: string[]
+          shouldError: boolean | undefined
+        }
+      | undefined = {
+      overrideEntities: undefined,
+      overrideRoutes: undefined,
+      shouldError: false,
+    }
+  ): void {
+    const { overrideEntities, overrideRoutes, shouldError } = options
+    const entities = overrideEntities ?? []
+    if (!overrideEntities) {
+      if (auth.userEntity) {
+        entities.push(auth.userEntity)
+      }
+      if (auth.externalAuthEntity) {
+        entities.push(auth.externalAuthEntity)
+      }
+    }
+
+    const routes = overrideRoutes ?? []
+    if (!overrideRoutes) {
+      if (auth.methods.email?.emailVerification.clientRoute) {
+        routes.push(auth.methods.email.emailVerification.clientRoute)
+      }
+      if (auth.methods.email?.passwordReset.clientRoute) {
+        routes.push(auth.methods.email.passwordReset.clientRoute)
+      }
+    }
+
+    const entityRefParser = makeRefParser('Entity', entities)
+    const routeRefParser = makeRefParser('Route', routes)
+
+    if (shouldError) {
+      expect(() =>
+        mapAuth(auth, entityRefParser, routeRefParser)
+      ).toThrowError()
+      return
+    }
+
+    const result = mapAuth(auth, entityRefParser, routeRefParser)
+
+    expect(result).toStrictEqual({
+      userEntity: entityRefParser(auth.userEntity),
+      externalAuthEntity:
+        auth.externalAuthEntity === undefined
+          ? undefined
+          : entityRefParser(auth.externalAuthEntity),
+      methods: mapAuthMethods(auth.methods, routeRefParser),
+      onAuthFailedRedirectTo: auth.onAuthFailedRedirectTo,
+      onAuthSucceededRedirectTo: auth.onAuthSucceededRedirectTo,
+      onBeforeSignup: auth.onBeforeSignup && mapExtImport(auth.onBeforeSignup),
+      onAfterSignup: auth.onAfterSignup && mapExtImport(auth.onAfterSignup),
+      onAfterEmailVerified:
+        auth.onAfterEmailVerified && mapExtImport(auth.onAfterEmailVerified),
+      onBeforeOAuthRedirect:
+        auth.onBeforeOAuthRedirect && mapExtImport(auth.onBeforeOAuthRedirect),
+      onBeforeLogin: auth.onBeforeLogin && mapExtImport(auth.onBeforeLogin),
+      onAfterLogin: auth.onAfterLogin && mapExtImport(auth.onAfterLogin),
+    } satisfies AppSpec.Auth)
+  }
+})
+
+describe('mapAuthMethods', () => {
+  test('should map minimal config correctly', () => {
+    testMapAuthMethods(Fixtures.getAuthMethods('minimal'))
+  })
+
+  test('should map full config correctly', () => {
+    testMapAuthMethods(Fixtures.getAuthMethods('full'))
+  })
+
+  test('should throw if emailVerification clientRoute ref is not provided when defined', () => {
```

Comment:
Ok, same thing with assumptions as I mentioned above. I will not mention it again but I am guessing there are other instances of this in this file.